geomesa-filter/src/test/scala/org/locationtech/geomesa/filter/FilterHelperTest.scala

```diff
@@ -195,6 +196,25 @@ class FilterHelperTest extends Specification {
       }
     }
 
+    "extract bounds from case-insensitive matches" >> {
+      def matcher(f: String, expected: Seq[String], rel: Option[String] => Option[String]): MatchResult[Any] = {
+        val res = FilterHelper.extractAttributeBounds(ECQL.toFilter(f), "name", classOf[String])
+        forall(res.values)(b => rel(b.lower.value) mustEqual b.upper.value)
+        res.values.map(_.lower.value.orNull) must containTheSameElementsAs(expected)
+      }
+      def literal(f: String, expected: Seq[String]): MatchResult[Any] = matcher(f, expected, o => o)
+      def wildcard(f: String, expected: Seq[String]): MatchResult[Any] =
+        matcher(f, expected, o => o.map(_ + WildcardSuffix))
+
+      literal("name ilike 'foo'", Seq("foo", "Foo", "fOo", "foO", "fOO", "FoO", "FOo", "FOO"))
+      wildcard("name ilike 'foo%'", Seq("foo", "Foo", "fOo", "foO", "fOO", "FoO", "FOo", "FOO"))
+      literal("name ilike 'f2o'", Seq("f2o", "F2o", "f2O", "F2O"))
+      wildcard("name ilike 'f2o%'", Seq("f2o", "F2o", "f2O", "F2O"))
+      // test max expansion limit of 10 chars
+      literal("name ilike 'fooooooooooo'", Seq.empty)
+      wildcard("name ilike 'fooooooooooo%'", Seq.empty)
+    }
```

Comment:
As an *incredibly picky* ask, should we add a unit test where the System property controlling the length of wildcard expansion is overridden?

Reply:
sure

Reply:
added